### PR TITLE
ultradns: improve set usage

### PR DESCRIPTION
**Pull request by josephholsten**

### CODE REVIEW

File: builtin/providers/ultradns/resource_ultradns_probe_http_test.go [hunk 1]
```diff
@@ -52,34 +52,34 @@ func TestAccUltradnsProbeHTTP(t *testing.T) {
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "zone", domain),
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "name", "test-probe-http-maximal"),
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "pool_record", "10.2.1.1"),
-					resource.TestCheckResourceAttr("ultradns_probe_http.it", "agents.0", "DALLAS"),
-					resource.TestCheckResourceAttr("ultradns_probe_http.it", "agents.1", "AMSTERDAM"),
+					resource.TestCheckResourceAttr("ultradns_probe_http.it", "agents.4091180299", "DALLAS"),
+					resource.TestCheckResourceAttr("ultradns_probe_http.it", "agents.2144410488", "AMSTERDAM"),
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "interval", "ONE_MINUTE"),
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "threshold", "1"),
 					resource.TestCheckResourceAttr("ultradns_probe_http.it", "http_probe.0.transaction.0.method", "POST"),
```
Comment (mlrobinson) [inline builtin/providers/ultradns/resource_ultradns_probe_http_test.go:59]: Should all these http_probe entries be re-indexed as well?

Comment (mlrobinson) [reply builtin/providers/ultradns/resource_ultradns_probe_http_test.go:59]: Resolved, or, Explained.

**Comment by mlrobinson** [comment]
LGTM

File: builtin/providers/ultradns/common.go [hunk 2]
```diff
@@ -155,11 +154,9 @@ func mapFromLimit(name string, l udnssdk.ProbeDetailsLimitDTO) map[string]interf
 
 // hashLimits generates a hashcode for a limits block
 func hashLimits(v interface{}) int {
-	var buf bytes.Buffer
 	m := v.(map[string]interface{})
-	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
-	h := hashcode.String(buf.String())
-	log.Printf("[INFO] hashLimits(): %v -> %v", buf.String(), h)
+	h := hashcode.String(m["name"].(string))
+	log.Printf("[INFO] hashLimits(): %v -> %v", m["name"].(string), h)
```
Comment (jmasseo) [inline builtin/providers/ultradns/common.go]: should this be debug?

File: builtin/providers/ultradns/resource_ultradns_dirpool.go [hunk 3]
```diff
@@ -544,26 +544,18 @@ func mapFromGeoInfos(gi *udnssdk.GeoInfo) []map[string]interface{} {
...
 	m := v.(map[string]interface{})
 	buf.WriteString(fmt.Sprintf("%s-", m["start"].(string)))
 	buf.WriteString(fmt.Sprintf("%s-", m["end"].(string)))
 	buf.WriteString(fmt.Sprintf("%s-", m["cidr"].(string)))
-	buf.WriteString(fmt.Sprintf("%s-", m["address"].(string)))
+	buf.WriteString(fmt.Sprintf("%s", m["address"].(string)))
 
-	return hashcode.String(buf.String())
+	h := hashcode.String(buf.String())
+	log.Printf("[INFO] hashIPInfoIPs(): %v -> %v", buf.String(), h)
```
Comment (jmasseo) [inline builtin/providers/ultradns/resource_ultradns_dirpool.go]: debug?

File: builtin/providers/ultradns/common.go [hunk 4]
```diff
@@ -155,11 +154,9 @@ func mapFromLimit(name string, l udnssdk.ProbeDetailsLimitDTO) map[string]interf
 
 // hashLimits generates a hashcode for a limits block
 func hashLimits(v interface{}) int {
-	var buf bytes.Buffer
 	m := v.(map[string]interface{})
-	buf.WriteString(fmt.Sprintf("%s-", m["name"].(string)))
```
Comment (jmasseo) [inline builtin/providers/ultradns/common.go:160]: why did this exist?  i mean i have ideas but curiosity...

**Review by jmasseo — APPROVED** [review]
bikeshedding and questions

File: builtin/providers/ultradns/common.go (hunk 4 above)
Comment (josephholsten) [reply builtin/providers/ultradns/common.go:160]: well, other things had multiple attributes concatenated together. This was a copy-paste.

File: builtin/providers/ultradns/common.go (hunk 2 above)
Comment (josephholsten) [reply builtin/providers/ultradns/common.go]: Yes, changed.

File: builtin/providers/ultradns/resource_ultradns_dirpool.go (hunk 3 above)
Comment (josephholsten) [reply builtin/providers/ultradns/resource_ultradns_dirpool.go]: Yes, changed.

**Pull request merged**